File: Programming-Basics-With-JavaScript/Homeworks-And-Labs/06.Nested-Loop/02-exercise/05.specialNumbers.js
function specialNumbers(input) {

    const num = Number(input[0]);

    let printLine = "";
    let isSpecial = false;

    for (let i = 1111; i <= 9999; i++) {
        let currentNum = i + "";
        for (let j = 0; j < currentNum.length; j++) {
            let currentDigit = Number(currentNum.charAt(j));
            if (num % currentDigit === 0) {
                isSpecial = true;
            } else {
                isSpecial = false;
                break;
            }
        }

        if (isSpecial) {
            printLine += currentNum + " ";
        }
    }

    console.log(printLine);
}
specialNumbers(["16"]);

File: Python_Basics/Homeworks-And-Labs/06.nested_loops/02.exercise/05.special_numbers.py
num = int(input())

print_line = ""
is_special = False

for i in range(1111, 9999 + 1):
    current_num = str(i)
    for j in range(len(current_num)):
        current_digit = int(current_num[j])
        if current_digit == 0:
            is_special = False
            break
        elif num % current_digit == 0:
            is_special = True
        else:
            is_special = False
            break
            
    if is_special:
        print_line += str(current_num) + " "
        
print(print_line)
